### ISSUE TITLE
Feature Request: Autosave

### ISSUE THREAD

**Issue by itsfarseen** [andmarti1424/sc-im#419]
First of all, thanks for the awesome sc-im :)
I use sc-im to track my time on a laptop without battery. 
Where I live, power cuts are quite frequent, multiple times per day. 
It would be really nice to have an autosave feature that saves the spreadsheet everytime I make a change.

**Comment by andmarti1424** [comment]
You have autobackup function.

**Comment by happycoder97** [comment]
There is a problem with autosave. It becomes empty when my computer loses power.
I am not really sure, but I think this is because of continuously rewriting the backup file every x seconds. 
This can be avoided by writing only once when I modify a file. Next backup should only happen after I modify it again.
Or it would be even nicer if the changes can be directly written to the actual file instead of the backup file.

**Comment by andmarti1424** [comment]
Hello @itsfarseen . Actually that shouldn't happen since the backup is made with a new name and once its finished the old backup is replaced with the new one.
There is a little little change that power goes down just after the delete and before that rename. See lines 1666 and 1669 of file.c. But if that is the case you should end up with a .new file with the backup.. and no .bak file.
You shouldnt have and empty file..

**Comment by itsfarseen** [comment]
I'm no longer facing this issue. Closing.
